Update lawn-mower.md with returning state (#2280)
home-assistant/developers.home-assistant: blog/2024-09-04-lawn-mower-activity-returning.md

---
author: Michael Arthur
authorURL: https://github.com/mikey0000
title: "New returning type in LawnMowerActivity"
---

As of Home Assistant Core 2024.9, integrations using the lawn_mower platform can indicate that their mower 

> is returning by using the new `LawnMowerActivity.RETURNING` state.

Reviewer: ```suggestion
is returning to the dock by using the new `LawnMowerActivity.RETURNING` state.
```

> is returning to the dock by using the new `LawnMowerActivity.RETURNING` state.

Reviewer: Please add a link to the dev docs page of the lawnmower entity.

https://developers.home-assistant.io/docs/core/entity/lawn-mower#activities

Author: done

> As of Home Assistant Core 2024.9, integrations using the lawn_mower platform can indicate that their mower 
> is returning to the dock by using the new `LawnMowerActivity.RETURNING` state.

Reviewer 2: ```suggestion
As of Home Assistant Core 2024.9, integrations implementing `LawnMowerEntity` can indicate that their mower is returning to the dock by using the new `LawnMowerActivity.RETURNING` state.
```

## Example

```python
    async def async_dock(self) -> None:
        """Start docking."""
        self._attr_activity = LawnMowerActivity.RETURNING
        self.async_write_ha_state()
```

> or returning `LawnMowerActivity.RETURNING` in the activity function property of your lawn_mower entity.

Reviewer 2: ```suggestion
or by returning `LawnMowerActivity.RETURNING` in the activity property of your `lawn_mower` entity.
```

> ---

Reviewer 2: Update the filename to todays date and then I think it's good to go 👍

Author: changed to 08-23

---
author: Michael Arthur
authorURL: https://github.com/mikey0000
title: "New returning type in LawnMowerActivity"
---

As of Home Assistant Core 2024.9, integrations implementing `LawnMowerEntity` can indicate that their mower is returning to the dock by using the new `LawnMowerActivity.RETURNING` state.

## Example

```python
    async def async_dock(self) -> None:
        """Start docking."""
        self._attr_activity = LawnMowerActivity.RETURNING
        self.async_write_ha_state()
```

or by returning `LawnMowerActivity.RETURNING` in the activity property of your `lawn_mower` entity.

```python
    @property
    def activity(self) -> LawnMowerActivity:
        """Return the state of the mower."""
        mower = self.mower
        if mower.state is RETURNING:
            return LawnMowerActivity.RETURNING
        ...
```

## More information

> More details can be found in the [documentation](https://developers.home-assistant.io/docs/core/entity/lawn-mower#activities)

Reviewer 2: Use relative linking and I think you could move this up to where the comment was made instead of here in the bottom.

Author: moved and made relative, probably want to make the previous blog post relative too then.
